### PR TITLE
Add Spotify provider

### PR DESCRIPTION
**Pull request by nickdirienzo**
Hi @singingwolfboy! I found out about `Flask-Dance` last night. It's a really great framework, nice work.

I'm working on a project using Spotify's APIs. Instead of keeping the provider implementation as custom, I figured to contribute it back upstream. I still need to add docs, but wanted to open this PR sooner than later, so I can stop using my fork in my requirements file.

Please let me know if there's anything you'd like me to change. Happy to include docs in this PR or follow-up separately. I didn't see a `CONTRIBUTING` file so I hope I'm adding a provider correctly and didn't miss anything (besides docs, heh).

### CODE REVIEW

**Comment by singingwolfboy** [comment]
Thanks for submitting this pull request! I haven't had time to review it yet, and I expect I won't for a few days still. I just wanted to add a comment so that you know I've seen it, and I _intend_ to get to it when I can!

**Comment by singingwolfboy** [comment]
Hey @nickdirienzo, sorry it took me so long to get to your pull request. This looks great! I actually documented of requirements for adding new providers in the [`flask_dance/contrib/README.rst`](https://github.com/singingwolfboy/flask-dance/blob/master/flask_dance/contrib/README.rst) file in the repository, and this pull request meets _almost_ all of them. As you pointed out, you're missing documentation, but fixing that should be as simple as adding this provider to the `docs/providers.rst` file. If you want to write a quickstart as well, that would be great, but not necessary. (As an aside, I'm aware that there's lots of repetition in the docs -- I wish there was a way to make Sphinx smart enough that it wasn't necessary!)

You also need to update the `CHANGELOG.rst` file: just add it to the "unreleased" section. Please do _not_ increment the version number -- I'll do that in a separate pull request, just before I actually make a new release and push it to PyPI.

I'm glad you find Flask-Dance useful! Let me know if there's anything else I can do to make it better. 😄

File: tests/contrib/test_spotify.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    assert client_id == "foo"
+
+
+@responses.activate
+def test_context_local():
+    responses.add(responses.GET, "https://google.com")
+
+    # set up two apps with two different set of auth tokens
+    app1 = Flask(__name__)
+    ghbp1 = make_spotify_blueprint(
```
Comment (singingwolfboy) [inline tests/contrib/test_spotify.py]: This is totally just a minor nitpick, but I'll mention it anyway. Looks like you copied these tests from the GitHub blueprint: `ghbp1` stands for "github blueprint 1". Could you modify this variable to `sbp1` (for "spotify blueprint 1") instead? It makes the tests a tiny bit more readable.

Comment (nickdirienzo) [reply tests/contrib/test_spotify.py]: Totally agree with you here on improving readability. Updated the variable names. Good catch! 👍

**Comment by nickdirienzo** [comment]
@singingwolfboy no worries on taking some time to get around to reviewing this; it took me a bit to respond to your comments and helpful feedback.

Looks like I totally missed the README inside of `contrib`. I went through that just now and pushed my changes up here. WRT Sphinx, I'm not sure how to make reST repeat items while changing a name from some variable; documentation tools are definitely a weak spot of mine.

Thanks for being a communicative maintainer. Keep up the great work!

**Comment by singingwolfboy** [comment]
Looks great! I'll merge this now, and cut a new release later today (or in the next few days, if I don't get time today). Once this code is out on PyPI, I'll add another comment to this pull request, so that you'll know to change your project to depend on the latest version of Flask-Dance!

**Comment by nickdirienzo** [comment]
Woohoo! Thanks @singingwolfboy!

**Comment by singingwolfboy** [comment]
Sorry it took me way longer than I thought, but [version 0.13.0 is finally out](https://pypi.python.org/pypi/Flask-Dance), and it includes your Spotify provider! :tada:

**Comment by nickdirienzo** [comment]
No worries at all. Thanks for releasing!

**Pull request merged**